src/message_filter.rs

```diff
@@ -15,10 +15,11 @@
 // Please review the Licences for the specific language governing permissions and limitations
 // relating to use of the SAFE Network Software.
 
-use std::collections::hash_map::DefaultHasher;
+use std::collections::{HashMap, VecDeque};
+use std::collections::hash_map::{DefaultHasher, Entry};
 use std::hash::{Hash, Hasher};
 use std::marker::PhantomData;
-use std::time::{Duration, SystemTime};
+use std::time::{Duration, Instant};
 
 fn hash<T: Hash>(t: &T) -> u64 {
```

Comment:
Yes, but both `Instant` and `SystemTime` break test determinism unless they are mocked.